content/agent/basic_agent_usage/docker.md

```diff
@@ -91,6 +91,10 @@ Exclude containers from the metrics collection and Autodiscovery, if these are n
 
 **Note**: The `docker.containers.running`, `.stopped`, `.running.total` and `.stopped.total` metrics are not affected by these settings and always count all containers. This does not affect your per-container billing.
 
+#### Misc
+
+- `DD_PROCESS_AGENT_CONTAINER_SOURCE`: Overrides container source auto-detection to force a single source. e.g `"docker"`, `"ecs_fargate"`, `"kubelet"`
```

Comment:
I think we'd want to put these in the docs for each specific integration vs here. If I want to install for fargate, for example, I end up https://docs.datadoghq.com/integrations/ecs_fargate/ or https://www.datadoghq.com/blog/monitor-aws-fargate/. The latter actually has the config file where we'd want to override the container source (although I'm curious if this is auto-detected cc @xvello?)

For kubelet I'd put it in our kubernetes setup docs.

@l0k0ms Any other ideas here?

Reply:
cc @mikezvi also for ideas

Reply:
Autodetection works for docker, ecs_fargate and kubelet, so I think we should not bother users for the usual case.
The edge case we need this option for is to force kubelet if docker is also available (as we give priority on the runtime over the orchestrator for continuity). I think that should just be listed on a "if the autodiscovered source is not what you want, you can override" basis